app/src/main.ts

```diff
@@ -30,6 +31,16 @@ if (require('electron-squirrel-startup')) {
 
 const appArgs = JSON.parse(fs.readFileSync(APP_ARGS_FILE_PATH, 'utf8'));
 
+// Do this relatively early so that we can start storid appData with the app
+if (appArgs.portable) {
+  log.debug(
+    'Setting appData and userData to ',
+    path.resolve(path.join(__dirname, '..', 'appData')),
+  );
+  app.setPath('appData', path.join(__dirname, '..', 'appData'));
+  app.setPath('userData', path.join(__dirname, '..', 'appData'));
```

Comment:
Is there any win to setting this to the same folder? Isn't there a risk of two files from `appData` & `userData` get overwritten? Why not

```suggestion
  app.setPath('appData', path.join(__dirname, '..', 'appData'));
  app.setPath('userData', path.join(__dirname, '..', 'userData'));
```

?

Reply:
Second question: does this work in Mac? In this case, it writes in the app bundle, correct?

Reply:
> Is there any win to setting this to the same folder? Isn't there a risk of two files from `appData` & `userData` get overwritten? Why not?

I don't think there is a real risk here, as `appData` is the "generic" where does data get stored directory, and `userData` is the where does MY data get stored. So under normal circumstances `userData` is a subfolder of `appData` (to keep its data separate from other apps), but in this context it shouldn't matter as we're sandboxed to this apps' context. If we want to be extra careful we could do:

```suggestion
  app.setPath('appData', path.join(__dirname, '..', 'appData'));
  app.setPath('userData', path.join(__dirname, '..', 'appData', app.name));
```

Or better yet instead of app.name, get the generated app name out of the package.json

> Second question: does this work in Mac? In this case, it writes in the app bundle, correct?

Yep, I verified this on my Mac. 😄

Reply:
> `appData` is the "generic" where does data get stored directory, and `userData` is the where does MY data get stored. So under normal circumstances `userData` is a subfolder of `appData` (to keep its data separate from other apps), but in this context it shouldn't matter as we're sandboxed to this apps' context. If we want to be extra careful we could do [...]

@TheCleric okay I get it. Verified on my local Linux box, it's only `userData` that is ever used, the `appData` folder is never even created. All good.